test/boulder-tools/docker-build.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -feuxo pipefail
+
+cd $(dirname $0)
+source go_ci_versions.sh
+
+docker build \
+  --build-arg "GO_VERSION=1.21.5" \
```

Comment:
Since you're sourcing `go_ci_versions.sh` which gives this script access to the `GO_CI_VERSIONS` array, is the plan to build a container for all those versions, or build only the newest version (presumably `${GO_CI_VERSIONS[0]}` or should we sort this?)?